src/generators.cpp

```diff
@@ -354,28 +395,32 @@ void Generator::GenerateNextToken() {
   }
 }
 
-DeviceSpan<int32_t> Generator::GetSequence(size_t index) const {
-  return search_->GetSequence(index);
+void Generator::RewindToLength(size_t new_length) {
+  if (model_->config_->model.type == "whisper" || model_->config_->model.type == "phi3v" || model_->config_->model.type == "decoder-pipeline")
+    throw std::runtime_error("RewindTo is currently not supported for " + model_->config_->model.type + ".");
+  if (new_length > search_->GetSequenceLength())
+    throw std::runtime_error("Cannot rewind to a length greater than the current sequence length");
+  if (new_length == search_->GetSequenceLength())
+    return;
```

Comment:
```suggestion
    return;  // Nothing to do, length doesn't change
```

What happens if you don't early exit on this check? If it works fine and it's not costly, we should just remove the check.

Reply:
It will set done to fasle. I think this is the best way to ensure it does nothing when we rewind to the current sequence length